connector/connect/src/main/scala/org/apache/spark/sql/connect/planner/SparkConnectPlanner.scala

```diff
@@ -123,6 +125,24 @@ class SparkConnectPlanner(plan: proto.Relation, session: SparkSession) {
     logical.Range(start, end, step, numPartitions)
   }
 
+  private def transformRenameColumns(rel: proto.RenameColumns): LogicalPlan = {
+    val input = new QueryExecution(session, transformRelation(rel.getInput)).analyzed
+    if (input.schema.size != rel.getColumnNamesCount) {
+      throw InvalidPlanInput(
+        "The number of columns doesn't match.\n" +
+          s"Old column names (${input.schema.size}): "
+          + input.schema.fields.map(_.name).mkString(", ") + "\n" +
+          s"New column names (${rel.getColumnNamesCount}): "
+          + rel.getColumnNamesList.asScala.mkString(", "))
+    }
+
+    val newCols =
+      input.output.zip(rel.getColumnNamesList.asScala).map { case (oldAttribute, newName) =>
+        Alias(oldAttribute, newName)()
+      }
+    logical.Project(newCols, input)
```

Comment:
i guess this can be simpler:
```
Dataset.ofRows(session, transformRelation(rel.getInput)).toDF(rel.getColumnNamesList.asScala).logicalPlan
```

Reply:
Correct me if I am wrong:

`toDF(String*)` has a varargs parameter and I found there is no way to pass a list/seq to it. We need update API to `toDF(String_*)`. Will this make it become a API change we probably shouldn't do it now?

Reply:
`toDF(rel.getColumnNamesList.asScala: _*)`

Reply:
Done thanks!